File: 520-detect-capital/520-detect-capital.java
class Solution {
    public boolean detectCapitalUse(String word) {
        if(word.toLowerCase().equals(word)) return true;
        if(word.toUpperCase().equals(word)) return true;
        int n = word.length();
        int sum = 1;
        if(Character.isUpperCase(word.charAt(0))){
            while(sum < n && Character.isLowerCase(word.charAt(sum))){
                sum++;
            }
        }
        return sum == n;
    }
}

File: 520-detect-capital/520-detect-capital.js
/**
 * @param {string} word
 * @return {boolean}
 */
var detectCapitalUse = function(word) {
    let pattern = /[A-Z]/;
    let sum = 0; 
    let index = 0;
    for(let i = 0; i<word.length; i++){
        if(pattern.test(word[i])){
            sum++;
            index = i;
        }
    }
    return sum === word.length || (sum === 1 && index === 0) || sum === 0;
};